store/sdk/src/main/java/org/apache/carbondata/sdk/file/CarbonReaderBuilder.java

```diff
@@ -115,11 +115,13 @@ public CarbonReaderBuilder withHadoopConf(Configuration conf) {
       hadoopConf = FileFactory.getConfiguration();
     }
     CarbonTable table;
-    if (filterExpression != null) {
-      table = CarbonTable.buildTable(tablePath, tableName, hadoopConf);
-    } else {
-      table = CarbonTable.buildDummyTable(tablePath);
-    }
+    // now always infer schema. TODO:Refactor in next version.
+    table = CarbonTable.buildTable(tablePath, tableName, hadoopConf);
+    //    if (filterExpression != null) {
```

Comment:
If it is useless, just remove them. We are using git, no need to comment the useless code

Reply:
yeah. OK